src/library/Cardano/CoinSelection.hs

```diff
@@ -226,51 +271,98 @@ sumOutputs =  coinMapValue . outputs
 sumChange :: CoinSelection i o -> Coin
 sumChange = mconcat . change
 
--- | Represents a set of options to be passed to a coin selection algorithm.
+-- | Defines an __inclusive upper bound__ on the /number/ of inputs that
+--   a 'CoinSelectionAlgorithm' is allowed to select.
 --
-data CoinSelectionOptions i o e = CoinSelectionOptions
-    { maximumInputCount
+newtype CoinSelectionLimit = CoinSelectionLimit
+    { calculateLimit
         :: Word8 -> Word8
```

Comment:
It occurred to me that some blockchains might permit more than 255 tx inputs.
But that's for another PR I suppose.

Reply:
> It occurred to me that some blockchains might permit more than 255 tx inputs.

Great point!

I would also like to change this, though the `idealBatchSize` function from `Migration` currently depends on this type. If that function can be changed to work with a wider type (in a future PR), then we can raise this limit.